source/wp-content/themes/wporg-developer-2023/src/code-related/block.php

```diff
@@ -25,16 +33,95 @@ function init() {
  * @return string Returns the block markup.
  */
 function render() {
+	$uses        = null;
+	$used_by     = null;
+	$has_uses    = false;
+	$has_used_by = false;
+
+	if ( post_type_has_uses_info() ) {
+		$uses     = get_uses();
+		$has_uses = $uses->have_posts();
+	}
+
+	if ( post_type_has_usage_info() ) {
+		$used_by     = get_used_by();
+		$has_used_by = $used_by->have_posts();
+	}
+
+	if ( ! $has_uses && ! $has_used_by ) {
+		return '';
+	}
+
 	$title_block = sprintf(
 		'<h2 class="wp-block-heading">%s</h2>',
 		__( 'Related', 'wporg' )
 	);
 
 	$wrapper_attributes = get_block_wrapper_attributes();
 	return sprintf(
-		'<section %s>%s %s</section>',
+		'<section %s>%s %s %s</section>',
```

Comment:
Could you use numbered placeholders here?

```suggestion
		'<section %1$s>%2$s %3$s %4$s</section>',
```